### DISCUSSION TITLE
Nano Bridge SQL Caching not working

### DISCUSSION THREAD

**Discussion by parkershute** [nanomq/nanomq#1440] [Q&A] [answered]
I have a Nano bridge connected to an EMQX broker through a HAProxy Load Balancer. I am trying to test the caching through SQL. I have it configured to write to disk but nothing happens or I get a success message that the message was sent to EMQX even when I have already received the disconnect from EMQX. Is there a step I am missing? I need to buffer messages when a disconnect occurs between the two brokers.
My test is setup like: 
- Establish the bridge to the EMQX
- pub to the Nano using MQTTX, 
- sub to the EMQX broker
- The message is delivered as expected
- Then I kill the HAProxy service and I see the disconnect in the logs on the Nano broker
- I try and pub again from MQTTX and check the server for where the message is configured to be stored
- nothing is there and no message is delivered upon re-establishing the connection 

My version output: NanoMQ Messaging Engine for Edge Computing & Messaging bus v0.19.5-2

Here is my config:

  ````
bridges.mqtt.emqx {
        server = "tls+mqtt-tcp://mysrvr:8883"
        proto_ver = 5
        clientid="desk_bridge_client"
        keepalive = 60s
        clean_start = false
        username = "myuser"
        password = "mypsswd"

        ssl {
                keyfile = "my key"
                certfile = "my cert"
                cacertfile = "my ca"
        }
        forwards = ["#"]
 
}

bridges.mqtt.cache {
        ## Max message limitation for caching
        disk_cache_size = 102400

        ## Mounted file path
        mounted_file_path="/opt/"

        ## The threshold of flushing messages to flash
        flush_mem_threshold = 100
}
````

**Comment by JaylinYu** [comment]
Is there a sqlite db file in the path  "/opt/" as you configured?
As Last time I checked, it worked. either the SQLite is not enabled in your version, or nanomq has no writing authority to create DB file.

Perhaps it is related to the session keeping flag

Will check again

**Reply by parkershute** [reply, marked as answer]
The issue appears to be that I do not have the correct version of nano, as this was not the SQLite version.
